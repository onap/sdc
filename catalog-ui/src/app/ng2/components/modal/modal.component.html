<div class="custom-modal {{size}}">
    <div class="ng2-modal-content">
        <div class="ng2-modal-header">
            <span class="title">{{ title }}</span>
            <span class="close-button" (click)="close()"></span>
        </div>
        <div class="ng2-modal-body">
            <ng-content></ng-content>
        </div>
        <div class="ng2-modal-footer">
            <button *ngFor="let buttonName of buttonsNames"
                    class="tlv-btn {{buttons[buttonName].cssClass}}"
                    [disabled] = "buttons[buttonName].getDisabled && buttons[buttonName].getDisabled()"
                    (click) = "buttons[buttonName].callback()">{{buttons[buttonName].text}}</button>
        </div>
    </div>
</div>
<div class="modal-background"></div>
