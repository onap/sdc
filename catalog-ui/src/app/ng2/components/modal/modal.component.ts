/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

/**
 * Created by rc2122 on 6/1/2017.
 */
import { Component, ElementRef, Input, OnInit, OnDestroy } from '@angular/core';
import * as $ from 'jquery';
import {ButtonsModelMap} from "app/models/button";

@Component({
    selector: 'modal',
    templateUrl: './modal.component.html',
    styleUrls:['modal.component.less']
})

export class ModalComponent implements OnInit, OnDestroy {
    @Input() size: string; 'xl|l|md|sm|xsm'
    @Input() title: string;
    @Input() public buttons:ButtonsModelMap;
    private modalElement: JQuery;
    private buttonsNames:Array<string>;

    constructor( el: ElementRef ) {
        this.modalElement = $(el.nativeElement);
    }

    ngOnInit(): void {
        let modal = this;
        this.modalElement.appendTo('body');
        if(this.buttons){
            this.buttonsNames = Object.keys(this.buttons);
        }
    }

    ngOnDestroy(): void {
        this.modalElement.remove();
    }

    open(): void {
        this.modalElement.show();
        $('body').addClass('modal-open');
    }

    close(): void {
        this.modalElement.hide();
        $('body').removeClass('modal-open');
    }
}
