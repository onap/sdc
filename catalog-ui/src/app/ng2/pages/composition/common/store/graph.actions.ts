export enum SelectedComponentType {
    COMPONENT_INSTANCE = "COMPONENT_INSTANCE",
    GROUP = "GROUP",
    POLICY = "POLICY",
    TOPOLOGY_TEMPLATE = "TOPOLOGY_TEMPLATE"
}

export class UpdateSelectedComponentAction {
    static readonly type = '[COMPOSITION] UpdateSelectedComponent';

        constructor(public payload: {uniqueId?: string, type?: string}) {
    }
}
    
export class SetSelectedComponentAction {
    static readonly type = '[COMPOSITION] SetSelectedComponent';

    constructor(public payload: {component?: any, type?: SelectedComponentType}) {
        }
}

export class OnSidebarOpenOrCloseAction {
    static readonly type = '[COMPOSITION] OnSidebarOpenOrCloseAction';

    constructor() {
    }
}

export class TogglePanelLoadingAction {
    static readonly type = '[COMPOSITION] TogglePanelLoading';
    constructor(public payload: { isLoading: boolean}) {
    }
}
