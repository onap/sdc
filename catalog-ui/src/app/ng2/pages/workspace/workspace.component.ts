/**
 * Created by ob0695 on 6/11/2018.
 */
