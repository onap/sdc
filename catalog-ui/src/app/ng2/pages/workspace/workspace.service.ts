/**
 * Created by ob0695 on 6/5/2018.
 */
/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

/**
 * Created by rc2122 on 5/23/2017.
 */
import { Injectable } from '@angular/core';
import {WorkspaceMode, ComponentState, Role} from "../../../utils/constants";
import {Component as TopologyTemplate, ComponentMetadata} from "app/models";
import {CacheService} from "../../services/cache.service";
import {IComponentMetadata} from "../../../models/component-metadata";
import {ComponentType} from "../../../utils";

@Injectable()
export class WorkspaceService {

    public metadata:ComponentMetadata;

    constructor(private cacheService:CacheService) {

    }

    public setComponentMetadata = (metadata: ComponentMetadata) => {
        this.metadata = metadata;
    }

    public getMetadataType(): string {
        switch (this.metadata.componentType) {
            case ComponentType.SERVICE:
                return ComponentType.SERVICE;
            default:
                return this.metadata.resourceType;
        }
    }

    public getComponentMode = (component:TopologyTemplate):WorkspaceMode => {//return if is edit or view for resource or service
        let mode = WorkspaceMode.VIEW;

        let user = this.cacheService.get("user");
        if (component.lifecycleState === ComponentState.NOT_CERTIFIED_CHECKOUT &&
            component.lastUpdaterUserId === user.userId) {
            if ((component.isService() || component.isResource()) && user.role == Role.DESIGNER) {
                mode = WorkspaceMode.EDIT;
            }
        }
        return mode;
    }
}

  