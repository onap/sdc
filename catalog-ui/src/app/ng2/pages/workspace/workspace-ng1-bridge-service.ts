/**
 * Created by ob0695 on 6/24/2018.
 */
/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */
import {Store} from "@ngxs/store";
import {Injectable} from "@angular/core";
import {UpdateIsViewOnly} from "../../store/actions/workspace.action";

@Injectable()
export class WorkspaceNg1BridgeService {

    constructor(private store: Store) {
    };

    public updateIsViewOnly = (isViewOnly: boolean):void => {
        this.store.dispatch(new UpdateIsViewOnly(isViewOnly));
    }

}
