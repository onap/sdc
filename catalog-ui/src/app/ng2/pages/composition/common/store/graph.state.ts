import { Action, Selector, State, StateContext} from '@ngxs/store';
import {
    OnSidebarOpenOrCloseAction,
    SelectedComponentType,
    SetSelectedComponentAction,
    TogglePanelLoadingAction
} from "./graph.actions";
import { PolicyInstance, GroupInstance, Component as TopologyTemplate, ComponentInstance, LeftPaletteComponent, FullComponentInstance} from "app/models";
import { TopologyTemplateService } from "app/ng2/services/component-services/topology-template.service";
import { tap } from "rxjs/operators";
import { CompositionService } from "app/ng2/pages/composition/composition.service";
import {GroupsService} from "../../../../services/groups.service";
import {PoliciesService} from "../../../../services/policies.service";
import {WorkspaceService} from "../../../workspace/workspace.service";

export class CompositionStateModel {

    isViewOnly?: boolean;
    panelLoading?: boolean;
    selectedComponentType?: SelectedComponentType;
    selectedComponent?: PolicyInstance | GroupInstance | TopologyTemplate | ComponentInstance;
    withSidebar?: boolean;
}

@State<CompositionStateModel>({
    name: 'composition',
    defaults: {
         withSidebar: true
    }
})
export class GraphState {

    constructor(private topologyTemplateService: TopologyTemplateService,
                private compositionService: CompositionService,
         private policiesService:PoliciesService, private groupsService:GroupsService,
                 private workspaceService: WorkspaceService) {}

    @Action(SetSelectedComponentAction)
    setSelectedComponent({dispatch, getState, patchState}:StateContext<CompositionStateModel>, action: SetSelectedComponentAction) {
        
        const state:CompositionStateModel = getState();
        
        patchState({ panelLoading: true });

        if(action.payload.component instanceof ComponentInstance){
            let originComponent = this.compositionService.getOriginComponentById(action.payload.component.getComponentUid());
            if(!originComponent) {
                return this.topologyTemplateService.getFullComponent(action.payload.component.originType, action.payload.component.getComponentUid())
                .pipe(tap(resp => {
                    this.compositionService.addOriginComponent(resp);
                    this.compositionService.setSelectedComponentType(SelectedComponentType.COMPONENT_INSTANCE);
                    patchState({
                        selectedComponent: new FullComponentInstance(action.payload.component, resp), 
                        selectedComponentType: action.payload.type,
                        panelLoading: false
                    });
                    }, err => {
                        patchState({
                            panelLoading: false
                        })
                    }
                ));
            } else {
                patchState({
                    selectedComponent: new FullComponentInstance(action.payload.component, originComponent), 
                    selectedComponentType: action.payload.type,
                    panelLoading: false
                });
            }
        } else if (action.payload.component instanceof PolicyInstance) {
            let topologyTemplate = this.workspaceService.metadata;
            return this.policiesService.getSpecificPolicy(topologyTemplate.componentType, topologyTemplate.uniqueId, action.payload.component.uniqueId).pipe(tap(resp => 
                {
                    this.compositionService.updatePolicy(resp);
                    patchState({
                        selectedComponent: resp, 
                        selectedComponentType: action.payload.type,
                        panelLoading: false
                    })
                }, err => {
                    patchState({
                        panelLoading: false
                    })
                 }
            ));
        
        } else if (action.payload.component instanceof GroupInstance) {
            let topologyTemplate = this.workspaceService.metadata;
            return this.groupsService.getSpecificGroup(topologyTemplate.componentType, topologyTemplate.uniqueId, action.payload.component.uniqueId).pipe(tap(resp => {
                this.compositionService.updateGroup(resp);
                patchState({
                    selectedComponent: resp,
                    selectedComponentType: action.payload.type,
                    panelLoading: false
                });
            }, err => {
                patchState({
                    panelLoading: false
                })
             }
            ));
        } else { //TopologyTemplate
            patchState({
                selectedComponent: action.payload.component,
                selectedComponentType: action.payload.type,
                panelLoading: false
            })
        }
    }


    // @Action(UpdateSelectedComponentNameAction)
    // UpdateSelectedComponentNameAction({patchState}:StateContext<CompositionStateModel>, action: UpdateSelectedComponentNameAction) {

    //     switch(action.payload.type){
    //         case SelectedComponentType.COMPONENT_INSTANCE:
    //             this.store.dispatch(new UpdateComponentInstancesAction([action.payload.component]));
    //             break;
    //         case SelectedComponentType.POLICY:
    //             this.store.dispatch(new UpdatePolicyNameAction(action.payload.uniqueId, action.payload.newName));
    //             break;
    //         case SelectedComponentType.GROUP:
    //             this.store.dispatch(new UpdateGroupInstancesAction)

    //     }
    //     if(action.payload.type === SelectedComponentType.COMPONENT_INSTANCE){
            
    //     } 
    
    // }

    @Selector() 
    static getSelectedComponent(state:CompositionStateModel) {
        return state.selectedComponent;
    }

    @Selector() 
    static getSelectedComponentId(state:CompositionStateModel) {
        return state.selectedComponent.uniqueId;
    }

    @Selector() 
    static getSelectedComponentType(state:CompositionStateModel) {
        return state.selectedComponentType;
    }


    @Action(OnSidebarOpenOrCloseAction)
    onSidebarOpenOrCloseAction({getState, setState}:StateContext<CompositionStateModel>) {
        const state:CompositionStateModel = getState();

        setState({
            ...state,
            withSidebar: !state.withSidebar
        });
    }
    
    @Action(TogglePanelLoadingAction)
    TogglePanelLoading({patchState}:StateContext<CompositionStateModel>, action: TogglePanelLoadingAction) {

        patchState({
            panelLoading: action.payload.isLoading
        });
    }

    @Selector() static withSidebar(state):boolean {
        return state.withSidebar;
    }

}