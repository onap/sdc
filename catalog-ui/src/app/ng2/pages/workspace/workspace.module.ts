/**
 * Created by ob0695 on 6/4/2018.
 */
/**
 * Created by ob0695 on 6/4/2018.
 */
import {NgModule} from "@angular/core";
import {CompositionPageModule} from "../composition/composition-page.module";

import {NgxsModule} from "@ngxs/store";
import {TopologyTemplateService} from "../../services/component-services/topology-template.service";
import {WorkspaceState} from "../../store/states/workspace.state";
import {WorkspaceService} from "./workspace.service";
import {DeploymentPageModule} from "./deployment/deployment-page.module";
import {ToscaArtifactPageModule} from "./tosca-artifacts/tosca-artifact-page.module";
import {InformationArtifactPageModule} from "./information-artifact/information-artifact-page.module";
import { reqAndCapabilitiesModule } from "./req-and-capabilities/req-and-capabilities.module";
import {AttributesModule} from "./attributes/attributes.module";
import {ArtifactsState} from "../../store/states/artifacts.state";
import {InstanceArtifactsState} from "../../store/states/instance-artifacts.state";
import {DeploymentArtifactsPageModule} from "./deployment-artifacts/deployment-artifacts-page.module";
import { DistributionModule } from './disribution/distribution.module';
import { ActivityLogModule } from './activity-log/activity-log.module';

@NgModule({
    declarations: [],
    imports: [
        DeploymentPageModule,
        CompositionPageModule,
        AttributesModule,
        reqAndCapabilitiesModule,
        ToscaArtifactPageModule,
        DeploymentArtifactsPageModule,
        InformationArtifactPageModule,
        DistributionModule,
        ActivityLogModule,
        NgxsModule.forFeature([WorkspaceState, ArtifactsState, InstanceArtifactsState])
    ],

    exports: [],
    entryComponents: [],
    providers: [TopologyTemplateService, WorkspaceService]
})

export class WorkspaceModule {

    constructor() {

    }
}
