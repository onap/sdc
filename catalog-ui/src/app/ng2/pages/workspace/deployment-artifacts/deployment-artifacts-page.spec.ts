// import ' rxjs/add/observable/of';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture } from '@angular/core/testing';
import { NgxsModule, Store } from '@ngxs/store';
import { NgxDatatableModule } from '@swimlane/ngx-datatable';
import { SdcUiServices } from 'onap-ui-angular';
import { Observable } from 'rxjs/Observable';
import { deploymentArtifactMock } from '../../../../../jest/mocks/artifacts-mock';
import { ConfigureFn, configureTests } from '../../../../../jest/test-config.helper';
import { ComponentMetadata } from '../../../../models/component-metadata';
import { ArtifactsService } from '../../../components/forms/artifacts-form/artifacts.service';
import { CacheService } from '../../../services/cache.service';
import { TopologyTemplateService } from '../../../services/component-services/topology-template.service';
import { TranslateModule } from '../../../shared/translator/translate.module';
import { TranslateService } from '../../../shared/translator/translate.service';
import { ArtifactsState } from '../../../store/states/artifacts.state';
import { WorkspaceService } from '../workspace.service';
import { DeploymentArtifactsPageComponent } from './deployment-artifacts-page.component';
import {ModalService} from "../../../services/modal.service";

describe('deployment artifacts page', () => {

    let fixture: ComponentFixture<DeploymentArtifactsPageComponent>;
    let topologyTemplateServiceMock: Partial<TopologyTemplateService>;
    let workspaceServiceMock: Partial<WorkspaceService>;
    let loaderServiceMock: Partial<SdcUiServices.LoaderService>;
    let store: Store;

    beforeEach(
        async(() => {

            topologyTemplateServiceMock = {
                getArtifactsByType: jest.fn().mockImplementation((componentType, id, artifactType) => Observable.of(deploymentArtifactMock))
            };
            workspaceServiceMock = {
                metadata: <ComponentMetadata>{
                    uniqueId: 'service_unique_id',
                    componentType: 'SERVICE'
                }
            }

            loaderServiceMock = {
                activate: jest.fn(),
                deactivate: jest.fn()
            }
            const configure: ConfigureFn = (testBed) => {
                testBed.configureTestingModule({
                    declarations: [DeploymentArtifactsPageComponent],
                    imports: [NgxDatatableModule, TranslateModule, NgxsModule.forRoot([ArtifactsState])],
                    schemas: [NO_ERRORS_SCHEMA],
                    providers: [
                        {provide: WorkspaceService, useValue: workspaceServiceMock},
                        {provide: TopologyTemplateService, useValue: topologyTemplateServiceMock},
                        {provide: SdcUiServices.LoaderService, useValue: loaderServiceMock},
                        {provide: ArtifactsService, useValue: {}},
                        {provide: SdcUiServices.PopoverService, useValue: {}},
                        {provide: CacheService, useValue: {}},
                        {provide: SdcUiServices.ModalService, useValue: {}},
                        {provide: ModalService, useValue: {}},
                        {provide: TranslateService, useValue: {}}
                    ],
                });
            };

            configureTests(configure).then((testBed) => {
                fixture = testBed.createComponent(DeploymentArtifactsPageComponent);
                store = testBed.get(Store);
            });
        })
    );

    it('should match current snapshot of informational artifact pages component', () => {
        expect(fixture).toMatchSnapshot();
    });

    it('should see exactly 2 tosca artifacts', () => {
        fixture.componentInstance.ngOnInit();
        fixture.componentInstance.deploymentArtifacts$.subscribe((artifacts) => {
            expect(artifacts.length).toEqual(8);
        })
        store.selectOnce((state) => state.artifacts.deploymentArtifacts).subscribe((artifacts) => {
            expect(artifacts.length).toEqual(8);
        });
    });

});
