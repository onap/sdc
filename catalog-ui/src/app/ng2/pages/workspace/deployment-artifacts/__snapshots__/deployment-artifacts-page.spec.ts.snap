// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deployment artifacts page should match current snapshot of informational artifact pages component 1`] = `
<deployment-artifact-page
  addOrUpdateArtifact={[Function Function]}
  artifactsService={[Function Object]}
  cacheService={[Function Object]}
  deleteArtifact={[Function Function]}
  getEnvArtifact={[Function Function]}
  modalService={[Function Object]}
  openGenericArtifactBrowserModal={[Function Function]}
  openPopOver={[Function Function]}
  popoverContentComponent="undefined"
  popoverService={[Function Object]}
  sortArtifacts={[Function Function]}
  store={[Function Store]}
  table="undefined"
  translateService={[Function Object]}
  updateEnvParams={[Function Function]}
  workspaceService={[Function Object]}
>
  
</deployment-artifact-page>
`;
